src/cli/cmd/generate-delegator.report.ts

```diff
@@ -0,0 +1,56 @@
+import fs from "fs";
+import { join } from "path";
+
+import { readPaymentReport, writeDelegatorReport } from "src/fs-client";
+import { globalCliOptions } from "src/cli/global";
+
+import {
+  REPORTS_SUCCESS_PAYMENTS_DIRECTORY,
+  CUSTOM_DELEGATOR_REPORT_DIRECTORY,
+} from "src/utils/constants";
+
+import {
+  BasePayment,
+  DelegatorPayment,
+  EPaymentType,
+} from "src/engine/interfaces";
+import { sortBy } from "lodash";
+
+export const generateDelegatorReport = async (commandOptions) => {
+  await fs.readdir(REPORTS_SUCCESS_PAYMENTS_DIRECTORY, async (err, files) => {
+    const { startCycle, endCycle, delegator } = commandOptions;
+
+    let payments: (BasePayment | DelegatorPayment)[] = [];
+
+    const cycleSet = files
+      .map((n) => Number(n.replace(".csv", "")))
+      .filter((n) => n >= startCycle)
+      .filter((n) => n <= endCycle);
+
+    for (const cycle of cycleSet) {
+      const report = await readPaymentReport(
```

Comment:
Ideally we should handle situation when report is missing. (Possibly moved by baker 🤷🏻 )